### PR TITLE
stubs + unit tests for adaptorConstructor

### PR DESCRIPTION
**Pull request by dragos-rebegea**

### CODE REVIEW

File: relay/ethToElrond/bridgeExecutors/ethElrondBridgeExecutor_test.go [hunk 1]
```diff
@@ -1 +1,417 @@
...
+		assert.Nil(t, err)
+		assert.Equal(t, false, executor.HasPendingBatch())
+		executor.GetPendingBatch(nil)
+
+		assert.Equal(t, true, executor.HasPendingBatch())
+		assert.Equal(t, expected, executor.pendingBatch)
+	})
+}
+
+func TestLider(t *testing.T) {
```
Comment (iulianpascalau) [inline relay/ethToElrond/bridgeExecutors/ethElrondBridgeExecutor_test.go:104]: Leader* :D - valid on all occurrences

File: relay/ethToElrond/bridgeExecutors/ethElrondBridgeExecutor_test.go [hunk 2]
```diff
@@ -1 +1,417 @@
...
+			GetQuorumCalled: func(ctx context.Context) (uint, error) {
+				return 3, nil
+			},
+		}
+		executor, err := NewEthElrondBridgeExecutor(args)
+		assert.Nil(t, err)
+		ctx := context.Background()
+		assert.Equal(t, false, executor.IsQuorumReachedForProposeTransfer(ctx))
+	})
+	t.Run("less < quorum", func(t *testing.T) {
```
Comment (iulianpascalau) [inline relay/ethToElrond/bridgeExecutors/ethElrondBridgeExecutor_test.go:250]: :+1: for all edge cases tests

File: relay/ethToElrond/bridgeExecutors/mock/bridgeStub.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	if b.SignersCountCalled != nil {
+		return b.SignersCountCalled(ctx, id)
+	}
+	return 0
+}
+
+// -------- helper functions
+
+// incrementFunctionCounter increments the counter for the function that called it
+func (b *BridgeStub) incrementFunctionCounter() {
```
Comment (iulianpascalau) [inline relay/ethToElrond/bridgeExecutors/mock/bridgeStub.go:124]: not used?

File: relay/ethToElrond/bridgeExecutors/mock/quorumProviderStub.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package mock
+
+import "context"
+
+type QuorumProviderStub struct {
```
Comment (iulianpascalau) [inline relay/ethToElrond/bridgeExecutors/mock/quorumProviderStub.go:5]: missing comments on all exported items

File: relay/ethToElrond/bridgeExecutors/mock/topologyProviderStub.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package mock
+
+type TopologyProviderStub struct {
+	AmITheLeaderCalled func() bool
+	PeerCountCalled    func() int
+	CleanCalled        func()
+}
+
+func (s *TopologyProviderStub) AmITheLeader() bool {
```
Comment (iulianpascalau) [inline relay/ethToElrond/bridgeExecutors/mock/topologyProviderStub.go:9]: missing comments on all exported items

File: relay/ethToElrond/bridgeExecutors/mock/quorumProviderStub.go (hunk 4 above)
Comment (dragos-rebegea) [reply relay/ethToElrond/bridgeExecutors/mock/quorumProviderStub.go:6]: done
